### ISSUE TITLE
In High Contrast mode, Modal primary and secondary buttons do not show focus indicator

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#3883]
### Environment
OS: Windows, using high contrast (HC) mode
Browser: Firefox (adopts HC)
Carbon version": v10 React http://react.carbondesignsystem.com/iframe.html?id=modal--default

## Detailed description
The 'normal' presentation of the modal shows the focus indicator for each of the three tab stops when a user tabs through the modal dialog.  In High Contrast mode, only the Close button ('X') has a focus indicator, and only after the focus returns to the Close button.

###  What did you expect to happen?
Each of the three buttons should have a focus indicator that appears as the user tabs between them. The non-HC mode on Chrome can be used to illustrate.

When the modal first appears there is a blue line around the close button as a focus indicator (as shown in the below screenshot).
![image](https://user-images.githubusercontent.com/14298245/64045696-d08efd00-cb1e-11e9-8278-78ffd87be0ff.png)

Tabbbing to the Secondary button causes the focus indicator to appear there (in this case, through the slightly unusual method of both a blue and white stroke replacing the outer edge of the black button background). 
![image](https://user-images.githubusercontent.com/14298245/64041244-d7644280-cb13-11e9-99ab-80a32c339102.png)
Tabbing to the Primary button causes the focus indicator to appear there (in this case, a white stroke is added around a few pixels inside the outer edge of the blue background)
![image](https://user-images.githubusercontent.com/14298245/64041251-dcc18d00-cb13-11e9-989e-3b0dc4029507.png)

### What happened instead?
In HC mode, only the Close button has a focus indicator (and only noticeably when the focus wraps around).

When the dialog launches, there is no obvious indicator of focus. On VERY close scrutiny, when the user tabs once, 
![image](https://user-images.githubusercontent.com/14298245/64047074-945d9b80-cb22-11e9-89eb-6d823f734533.png)
there is a slight decrease in the thickness of the outline around the invisible close icon, as shown in the following two screen details. (Note: the X is invisible in HC mode. Already opened as issue #3881). 
![image](https://user-images.githubusercontent.com/14298245/64047080-9aec1300-cb22-11e9-9912-9414ecbba7d7.png)

There is no visible change to either the primary and secondary buttons when they receive focus
![image](https://user-images.githubusercontent.com/14298245/64039091-b4835f80-cb0e-11e9-96a5-29e6ddaafac0.png)

When the user tabs a third time and the focus wraps around to the (missing) Close icon, the focus indicator is not only the very hard to perceive difference in thickness, but now shows up with a more visible dotted (dashed, actually) focus rectangle  _inside_ the button outline for the X. 
![image](https://user-images.githubusercontent.com/14298245/64041109-76d50580-cb13-11e9-8271-e865ef51f63f.png)

Note that this dashed focus rectangle also appears in Firefox without HC mode.

### What WCAG 2.1 checkpoint does the issue violate?
Not clearly a violation, but obviously a DESIGN consideration. 

## Steps to reproduce the issue

1. From the Start menu, type High Contrast mode, and press Enter
1. Choose High Contrast #1
1. Launch the react carbon modal component in Firefox
1. Press F5 to refresh the browser (if it didn't adopt high contrast colours automatically)
1. Press Tab to move the focus between the buttons.

## Other Comments
Note that on Firefox in non-HC mode, when the focus wraps around to the Close icon, it shows an additional dashed line focus indicator. Although a designer might consider this an error, it is the only thing that works in HC mode. Without it, the user would never see a focus indicator.
![image](https://user-images.githubusercontent.com/14298245/64045960-6d519a80-cb1f-11e9-927c-3ec0923c25d9.png)

Another oddity is that if a Windows change event happens (i.e., the user switches tasks and puts another application window into the foreground on their system), when the browser page with the Carbon component is brought back to the foreground, the focus indicator around the close button disappears. Just pointing out as odd behaviour; not something I'm concerned with.
![image](https://user-images.githubusercontent.com/14298245/64046039-ad188200-cb1f-11e9-896a-fc71972d13f5.png)

**Comment by abbeyhrt** [comment]
Due to FireFox not supporting Microsoft's CSS extension for targeting elements in HCM, we officially do not support Windows & Firefox HCM. You can read about that decision [here]( https://github.com/carbon-design-system/carbon/blob/master/docs/on-windows-hcm-support.md). I opened up a different issue to address the dashed outline in firefox #3971
